diff --git a/dist/ghostty-web.js b/dist/ghostty-web.js
index a4ca427b95288e70c782ce303013cd644893dcec..dfe5b57ebb49a6df956b1eb1ea3c892b8999d293 100644
--- a/dist/ghostty-web.js
+++ b/dist/ghostty-web.js
@@ -2387,7 +2387,8 @@ class IA {
    */
   writeInternal(A, B) {
     var g;
-    this.wasmTerm.write(A), this.processTerminalResponses(), typeof A == "string" && A.includes("\x07") ? this.bellEmitter.fire() : A instanceof Uint8Array && A.includes(7) && this.bellEmitter.fire(), (g = this.linkDetector) == null || g.invalidateCache(), this.viewportY !== 0 && this.scrollToBottom(), typeof A == "string" && A.includes("\x1B]") && this.checkForTitleChange(A), B && requestAnimationFrame(B);
+    const E = this.viewportY, C = E !== 0 ? this.wasmTerm.getScrollbackLength() : 0;
+    this.wasmTerm.write(A), this.processTerminalResponses(), typeof A == "string" && A.includes("\x07") ? this.bellEmitter.fire() : A instanceof Uint8Array && A.includes(7) && this.bellEmitter.fire(), (g = this.linkDetector) == null || g.invalidateCache(), E !== 0 && this.scrollToLine(Math.max(0, Math.min(this.wasmTerm.getScrollbackLength(), E + Math.max(0, this.wasmTerm.getScrollbackLength() - C)))), typeof A == "string" && A.includes("\x1B]") && this.checkForTitleChange(A), B && requestAnimationFrame(B);
   }
   /**
    * Write data with newline
